Remove upload button

Anshiboy opened the pull request:
Features Added:

- The delete icon in ClaimsFileScreen now prompts a popup to confirm deletion
- Removed upload button from initial Files in NewClaimScreen

Things that can make it better (AND IN PROGRESS):

- Changing the PDF viewer to shop as a modal popup as it will give more freedom to view PDF

EDIT:
react bootstrap made the popup look aesthetically pleasing at the cost of removing all styles in the rest of the project. So for now I have disabled it until I find a way to safely isolate it only for the modal

Also, the code is up to date with Greg's latest PR pending approval

Make sure to 
npm install packages before.

geraldyywang commented on src/App.js:
Note that this reverts a bug fix in a previous pr that fixes front-end deployment. Please do not change the routing in this file!

geraldyywang commented on src/screens/ClaimDashboardScreen.js:
Good comments!

geraldyywang commented on src/screens/ClaimFilesScreen.js:
Good comments! Note that the modal does not close if the file is deleted. Please add this functionality!

geraldyywang reviewed (changes requested):
# Code Review
## Coding style
Good coding style and conventions!

## Correctness (checkout the branch and try running the code)
Good job on removing the upload button! However, I did notice some bugs with the current functionality:
1. When I click on the trash button and the confirmation pops up on the bottom, the dashboard disappears (see below)
![image](https://github.com/Flo-CI/front-end/assets/96800313/4c2e8e67-c335-4b92-b507-d18717f2eb29)
2. In App.js, the pr reverts the routing bug that was fixed so that the deployment runs.
![image](https://github.com/Flo-CI/front-end/assets/96800313/cbe7fed4-ca81-4c00-86da-da6ccf55a042)
3. After I click on 'Delete' for the confirmation, the confirmation does not disappear.

## Documentation
The documentation for the proposed changes (i.e. comments) are clear. Great job! 

## Clarity of pull request
This pull request's commit history is very clear. Good job! However, I'm slightly confused as to what has been actually implemented vs. what hasn't. Please also add some screenshots so that it's easier to understand what to look out for, and pr changes that are fully finished! Thanks :)

Anshiboy commented:
### Code update 2:

- The upload files button in the code allows to view and modify PDFs
- Upload button is still gone
- fixed all the bugs mentioned
- cleaned up the code a little and attempted to make it follow strategy design pattern

I have run tests locally to ensure the PDF viewer and upload was working, now just need to connect with backend. I will be writing tests shortly. 

Hope you like it :)

Pictures of code:
<img width="487" alt="image" src="https://github.com/Flo-CI/front-end/assets/87465536/a19a1e63-4d6e-4ceb-9172-53e5375c7514">

- Password

<img width="664" alt="image" src="https://github.com/Flo-CI/front-end/assets/87465536/8ac96758-2502-4f66-b54b-52112bc42ce1">

- upload files

<img width="1512" alt="image" src="https://github.com/Flo-CI/front-end/assets/87465536/9738a4a3-15f2-4201-8f60-8aca68ce5088">

- when uploading wrong format

<img width="1512" alt="image" src="https://github.com/Flo-CI/front-end/assets/87465536/1b1ef4e1-2fe5-4a04-83f4-eefea5240ece">

- pdf

<img width="602" alt="image" src="https://github.com/Flo-CI/front-end/assets/87465536/07b11b4a-01cc-4288-9e73-e677ceee1504">

- extra features when you scroll

LevOzay commented on src/screens/ClaimFilesScreen.js:
Except for the problem Gerald has mentioned the functionality looks fine. However, this code might not adhere to the React principles as you're implementing parts of the pdf viewer in the ClaimFilesScreen when you should be doing it inside DocumentViewer.

LevOzay commented on src/screens/DocumentViewer.js:
Why are you not exporting DocumentViewer? Is there another way you have connected it to the program?

LevOzay commented on src/screens/NewClaimScreen.js:
This has already been implemented. Please pull from main before you try to make any changes

sarvaSanjay commented on package-lock.json:
Why does your package-lock file have so many changes?

sarvaSanjay commented on src/App.css:
```diff
@@ -115,3 +115,11 @@ input:checked + .slider:before {
     transform: rotate(360deg);
   }
 }
+
```
Please replace this to use relative units (vh and vw). Using pixels makes the design non-responsive

sarvaSanjay commented on src/screens/ClaimDashboardScreen.js:
```diff
@@ -7,6 +7,17 @@ import NewClaimScreen from "./NewClaimScreen";
 import useAuthenticationCheck from "../hooks/useAuthenticationCheck.js";
 
+// Import Worker
+import { Worker } from '@react-pdf-viewer/core';
```
Why are pdf-viewers being imported into the ClaimDashboard, they aren't being used here

sarvaSanjay commented on src/screens/ClaimDashboardScreen.js:
This file is still displaying static content whereas in the `main` branch it is drawing from backend. This leads me to believe that you are not performing `git pull origin main` at your end as this branch is very clearly not up to date with the main branch. Please do this 🙏🏾🙏🏾🙏🏾

LevOzay reviewed (changes requested):
# Code Review
## Coding style
Good coding style and conventions!

## Correctness (checkout the branch and try running the code)
The correctness of the code you've written seems fine. However, there are many issues such as your branch not being up to date with main, DocumentViewer not being connected to the program, and the alteration of the routing which causes deployment errors. Please look into the individual comments I've written and make changes accordingly.

Yet, the biggest question mark about your changes is the alteration of package-lock up to thousands of lines. That most likely will affect everything in main which is a huge hazard. Please address this issue. One idea of resolving it might be pulling from main.
## Documentation

The documentation for the proposed changes are clear. Great job! 

## Clarity of pull request
This pull request's commit history and description outline the change made well. Good job!
